xonsh/completers/path.py

```diff
@@ -151,6 +151,59 @@ def _quote_paths(paths, start, end):
     return out
 
 
+def joinpath(path):
+    if path is None:
+        return ''
+    if len(path) == 0:
```

Comment:
I agree the bytecode will be identical (ish).  But then why use the `elif` below?  And do you even need the `else` or just a `return`?